### ISSUE TITLE
Update all outdated dependencies

### ISSUE THREAD

**Issue by amitksingh1490** [tailcallhq/rust-grpc#44]

**Comment by Nabanita29** [comment]
Is this issue still open?

**Comment by castrovroberto** [comment]
/attempt #44

<div id="algora-attempt" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/cm41swr8a0002jm03wcbo6jea/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by castrovroberto** [comment]
I'm having a significant issue trying to upgrade `hyper` dependency from `0.14.28` to `1.6.0` which is the latest available version at the moment. The problem I'm facing is that the server address bind logic has changed completely, so there is not a straight forward way to implement it without significant refactoring, which I'm not willing to continue pursuing at the moment. Feel free to pick up on that insight. I'm cancelling my attempt.
Regards.

**Comment by MustafaMulla29** [comment]
@amitksingh1490 is this open?

**Comment by jayantpranjal0** [comment]
@amitksingh1490 can you check out the existing pr. I would like to give the issue a shot, if the existing pr doesn't work.

**Comment by cinemaaiproductions-dev** [comment]
I'd like to work on this. Looking into it now.

**Comment by CuboYe** [comment]
/claim 44

Claiming territory for a scoped fix attempt now. If maintainer preference differs, I can adapt quickly.

**Comment by vegapunk9527** [comment]
/attempt #44

Plan: refresh the dependency set in small compile-safe steps instead of doing a blind version bump. I will inspect the current Cargo dependency graph, update outdated crates where API changes are mechanical, handle any required hyper/tonic/server API changes only if they can be kept narrow, run format/tests/build, and only open a PR with `/claim #44` if the updated workspace compiles cleanly.

**Comment by Christiantyemele** [comment]
is this still open , i want to claim it

**Comment by Christiantyemele** [comment]
Submitted PR #66 with semver-compatible dependency updates. 

Key updates:
- tokio 1.36.0 → 1.52.3
- anyhow 1.0.82 → 1.0.102
- tracing 0.1.40 → 0.1.44
- tracing-subscriber 0.3.18 → 0.3.23
- hyper 0.14.28 → 0.14.32
- gh-workflow 0.5.1 → 0.5.10
- once_cell 1.19.0 → 1.21.4
- prost/prost-types 0.12.3 → 0.12.6
- hyper-util 0.1.3 → 0.1.20
- http-body-util 0.1.0 → 0.1.3

All tests, fmt, and clippy checks pass. Breaking version updates (tonic, hyper 1.x, tower 0.5, opentelemetry, shuttle) are out of scope and require separate migration PRs.

**Comment by keremercin** [comment]
Hi, is this bounty still active and available to work on? If yes, I'd like to update the outdated dependencies and submit a PR.

**Comment by duongynhi000005-oss** [comment]
/attempt #44

Plan: audit the current Cargo dependency graph, update outdated crates in compile-safe steps, keep major breaking migrations out of scope unless they are mechanically small, and open a PR with validation once the workspace builds cleanly.

**Comment by CharlieLZ** [comment]
/attempt #44

Plan: submit a full current-major dependency migration, not just the semver-compatible lockfile bump already covered by other open PRs. This updates the Tonic/Prost stack to 0.14, Hyper to 1.x, Tower to 0.5, OpenTelemetry to 0.32, Shuttle to 0.57, updates the generated CI workflow, and adapts the source APIs so cargo test/fmt/clippy pass.

**Comment by zhangsiqiang519** [comment]
/attempt #44
Plan:
1) Bump outdated dependencies to latest compatible versions.
2) Fix compile/test fallout with minimal code changes.
3) Run full test suite and open PR with validation output.

**Comment by zhangsiqiang519** [comment]
Implemented in PR https://github.com/tailcallhq/rust-grpc/pull/69

- Updated outdated dependencies via `cargo update` (lockfile only)
- Verified with `cargo test`

/claim #44

**Comment by zhangsiqiang519** [comment]
Updated PR #69 with explicit Cargo.toml version bumps + lockfile refresh + passing cargo test. Ready for maintainer review and merge. /claim #44

**Comment by zhangsiqiang519** [comment]
PR #69 updated again with explicit dependency bumps, clippy/test validation, and a short demo artifact (`demo/issue-44-demo.gif`) per bounty guidelines. Ready for maintainer review. /claim #44

**Comment by Shaidyk** [comment]
/claim #44

PR: https://github.com/tailcallhq/rust-grpc/pull/73

**Comment by liukaisweetsmile-pixel** [comment]
/attempt #44

Implementation plan:
- Audit every outdated direct dependency and the lockfile, including major-version lines that compatible-only PRs leave unchanged.
- Migrate affected Hyper/Tonic/Prost/Tower/OpenTelemetry/Shuttle APIs in small, reviewable steps while preserving behavior.
- Update build/CI support where required, add or adjust targeted regression coverage, and avoid unrelated refactors.
- Validate formatting, workspace/all-feature checks, tests, and Clippy with warnings denied.
- Include a short demo artifact and a clear dependency/migration matrix in the PR.

If a major upgrade is blocked by an upstream incompatibility, I will document the exact blocker instead of hiding failures or claiming an incomplete migration.

**Comment by ChanghuLiu** [comment]
/attempt #44

Plan:
- Update the Rust dependency stack to current compatible releases.
- Migrate Tonic/Prost, OpenTelemetry, Shuttle, and gh-workflow APIs as needed.
- Refresh Cargo.lock to current compatible transitive versions.
- Preserve existing gRPC behavior and add regression tests for CRUD, batch lookup, and NotFound semantics.
- Validate with cargo test, cargo fmt, and cargo clippy.

**Comment by beefrea** [comment]
/attempt #44

Plan:
1. Audit outdated direct dependencies and lockfile
2. Upgrade within compatible ranges, fix breaking changes
3. Ensure build + tests pass
4. Open PR with `/claim #44`

**Comment by bilhokista** [comment]
Submitted PR resolving this conservatively: full Cargo.lock refresh to latest compatible versions, build+test green. Majors (tonic/hyper) excluded as breaking. PR link above.

**Comment by EEEEE213** [comment]
Is this bounty still funded and accepting new contributors? With several dependency-update PRs pending, what remains needed for acceptance? Is USDC payment on Base or Solana available?

**Comment by k4it00** [comment]
/attempt #44

Plan: run cargo update, upgrade the outdated direct deps to latest compatible semver, fix any breakages surfaced by cargo check/clippy/test, and open a PR with CI green. Will list every dep bump in the PR body.

**Comment by ossmydev** [comment]
/attempt #44

I’d like to work on this bounty.

Plan:

Audit the current Cargo dependencies and identify outdated packages.
Update dependencies incrementally to avoid unnecessary breaking changes.
Fix any compatibility issues introduced by the updates.
Run the full test suite and ensure the project still builds successfully.
Keep the PR focused only on dependency updates and required compatibility fixes.

**Comment by arthuz-th** [comment]
/attempt #44

Plan: refresh the repository's currently outdated direct dependencies to compatible current releases, keep the change dependency-focused, resolve any API breakages needed by those upgrades, and run the project's formatting/check/test suite before submitting. I will document the exact versions changed and verification scope, and include the required short demo evidence in the PR.

**Comment by stffinfcti** [comment]
/attempt #44

Plan: audit all direct dependencies against latest releases, perform the full major-line migration where the repo code permits (tonic/prost, hyper 1.x, tower, opentelemetry, shuttle, gh-workflow), adapt the affected server/reflection/telemetry call sites with minimal diffs, refresh Cargo.lock, and validate with cargo fmt, cargo clippy -D warnings, and cargo test before opening a PR with /claim #44.

**Comment by ananagila96-lang** [comment]
/attempt #44

Plan: audit the current Cargo.toml/Cargo.lock for outdated direct dependencies, update them in compatible groups, fix any API breakages required by the upgrades, then run fmt/clippy/tests/build and document the exact dependency changes and validation in the PR.
